src/NBitcoin/TransactionBuilder.cs

```diff
@@ -779,6 +779,22 @@ public TransactionBuilder Send(Script scriptPubKey, Money amount)
             return this;
         }
 
+        /// <summary>
+        /// Clears send builders so we can setup recipient sends again.
+        /// </summary>
+        public void ClearSendBuilders()
+        {
+            List<Builder> builders = this.CurrentGroup.Builders;
+            for (int i = builders.Count - 1; i >= 0; i--)
+            {
+                // All other builders have different return type.
+                if (builders[i].Method.ReturnType == typeof(Money))
+                {
+                    this.CurrentGroup.Builders.RemoveAt(i);
+                }
+            }
```

Comment:
Could be done in 1 line: 

```
this.CurrentGroup.Builders = this.CurrentGroup.Builders.Where(x => x.Method.ReturnType != typeof(Money)).ToList();
```

Reply:
Since method is called from just 1 place I guess it's better to replace method call with that one line

Reply:
I can't replace it as `CurrentGroup` is inaccessible from outside this class, hence I had to add this helper method.

Reply:
> Could be done in 1 line

Sure, but wouldn't it then involve 2 iterations: 1 for `Where` and one for `ToList` instead of just a simple `for` loop?

https://github.com/microsoft/referencesource/blob/master/System.Core/System/Linq/Enumerable.cs#L30
https://github.com/microsoft/referencesource/blob/master/System.Core/System/Linq/Enumerable.cs#L329